lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -458,11 +461,26 @@ private boolean achievedWithGpg45Profile(
                         ? gpg45ProfileEvaluator.getFirstMatchingProfile(
                                 gpg45Scores, requestedVot.getSupportedGpg45Profiles())
                         : Optional.empty();
+        List<SignedJWT> gpg45Credentials = new ArrayList<>();
+        for (SignedJWT credential : credentials) {
+            if (!VcHelper.isOperationalProfileVc(credential)) {
+                gpg45Credentials.add(credential);
+            }
+        }
 
         // Successful match
         if (matchedGpg45Profile.isPresent()) {
+            // remove weaker operational profile
+            if (configService.enabled(INHERITED_IDENTITY.getName())) {
+                for (SignedJWT credential : credentials) {
+                    if (VcHelper.isOperationalProfileVc(credential)) {
+                        verifiableCredentialService.deleteVcStoreItem(
+                                auditEventUser.getUserId(), HMRC_MIGRATION_CRI);
```

Comment:
Can we make this a bit more precise about what you're actually doing here - I think we're deleting any _HMRC_ inherited identity, and we should make the VC selection match that:

```
private void deleteHmrcInheritedIdentityIfPresent(List<VcStoreItem> vcStoreItems) {
    for (var vcStoreItem : vcStoreItems) {
        if (HMRC_MIGRATION_CRI.equals(vcStoreItem.getCredentialIssuer())) {
            verifiableCredentialService.deleteVcStoreItem(
                    vcStoreItem.getUserId(),
                    vcStoreItem.getCredentialIssuer());
        }
    }
}
```
I think this ensures that the thing being deleted is exactly what we're looking for. It also means that it won't pick up _other_ operational profiles, which might not want to be deleted.

Pulling it out as a separate method also allows you to implictly explain what's going on.

Reply:
This also matches the implementation you've got in `EvaluateGpg45ScoresHandler` :)

Reply:
I was thinking the same thing. But wasn't settled on the HMRC part yet. Which I wanted to discuss with you.